### PR TITLE
[terra-form-select-combobox] Added "Results that contain" text in combobox

### PR DESCRIPTION
**Pull request by gt106551**
### Summary
       Fixed-Added "Results that contain" text in combobox.

**What was changed:**
 Added  a  div tag to display "Results that contain" text in combobox
**Why it was changed:**
 Combobox should display a message in its selection menu, saying "Results that contain " while dynamically filtering options as the user types.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

### [UXPLATFORM-9733](https://jira2.cerner.com/browse/UXPLATFORM-9733)
---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-select/src/shared/_SearchResults.module.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:local {
+  .search-results {
+    color: #8d8d8d;
+    padding: 6px;
+    font-weight: 500;
+  }
+}
```
Comment (supreethmr) [inline packages/terra-form-select/src/shared/_SearchResults.module.scss:7]: Do we need the style changes ..?? As UX has not provided any style changes.

Comment (gt106551) [reply packages/terra-form-select/src/shared/_SearchResults.module.scss]: Removed styles and updated

File: packages/terra-form-select/src/shared/_SearchResults.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const propTypes = {
+  /**
+   * Content to display when search results are found.
+   */
+  searchResultContent: PropTypes.node,
+};
+
+const SearchResults = ({ searchResultContent }) => (
+  <div role="option" aria-selected="false">
```
Comment (supreethmr) [inline packages/terra-form-select/src/shared/_SearchResults.jsx:12]: I think we need to add styles for padding similar to no results component :
```css
    padding: var(--terra-form-select-search-results-padding, 0.28571rem 0.714286rem);
    word-break: break-word;
    word-wrap: break-word;
```

Comment (gt106551) [reply packages/terra-form-select/src/shared/_SearchResults.jsx]: Updated.

**Review by supreethmr — APPROVED** [review]

**Review by supreethmr — CHANGES_REQUESTED** [review]
Screenshot 2023-11-06 at 2 33 52 PM

No Matching Results screenshot is updated incorrectly !!

**Comment by gt106551** [comment]
> Screenshot 2023-11-06 at 2 33 52 PM
> No Matching Results screenshot is updated incorrectly !!

updated

File: packages/terra-form-select/src/orion-fusion-theme/SearchResults.module.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .orion-fusion-theme {
+    --terra-form-select-no-results-padding: 0.5rem 1rem;
```
Comment (supreethmr) [inline packages/terra-form-select/src/orion-fusion-theme/SearchResults.module.scss:3]:
```suggestion
   --terra-form-select-search-results-padding: 0.5rem 1rem;
```

File: packages/terra-form-select/src/clinical-lowlight-theme/SearchResults.module.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .clinical-lowlight-theme {
+    --terra-form-select-no-results-padding: 0.28571rem 0.714286rem;
```
Comment (supreethmr) [inline packages/terra-form-select/src/clinical-lowlight-theme/SearchResults.module.scss:3]:
```suggestion
    --terra-form-select-search-results-padding: 0.28571rem 0.714286rem;
```

File: packages/terra-form-select/src/orion-fusion-theme/SearchResults.module.scss (hunk 3 above)
Comment (gt106551) [reply packages/terra-form-select/src/orion-fusion-theme/SearchResults.module.scss]: Updated.

File: packages/terra-form-select/src/clinical-lowlight-theme/SearchResults.module.scss (hunk 4 above)
Comment (gt106551) [reply packages/terra-form-select/src/clinical-lowlight-theme/SearchResults.module.scss]: Updated.

**Review by supreethmr — APPROVED** [review]

**Review by saket2403 — APPROVED** [review]

**Comment by rahulkumar8cs** [comment]
+1

**Pull request merged**